toolkit/livedata/src/main/kotlin/br/com/arch/toolkit/livedata/_executor.kt

```diff
@@ -1,29 +1,59 @@
 package br.com.arch.toolkit.livedata
 
-import br.com.arch.toolkit.livedata.ExecutorUtil.async
 import br.com.arch.toolkit.livedata.response.MutableResponseLiveData
 import br.com.arch.toolkit.livedata.response.ResponseLiveData
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
 
 /**
  * Create a ResponseLiveData Instance posting Loading, Success or Error
  * All parameters are executes on the WorkerThread
  *
+ * @param scope The coroutine Scope to run this operation
  * @param errorTransformer To invoke the regular Exception caused by the param "func"
  * @param func Return the success value on the ResponseLiveData
  */
-inline fun <T> makeAsyncOperation(crossinline errorTransformer: (Exception) -> Exception, crossinline func: () -> T): ResponseLiveData<T> {
+inline fun <T> makeAsyncOperation(
+    scope: CoroutineScope,
+    crossinline errorTransformer: (Exception) -> Exception,
+    crossinline func: () -> T
```

Comment:
Já que está integrando com coroutines, seria interessante uma versão onde `func` fosse `suspend`

Reply:
faz sentido.

Reply:
Fiquei pensando nisso, na real eu to usando coroutine só pra quebrar thread, e isso é opcional... se eu colocar como suspend, eu vou ter que obrigar todo responseLiveData a ter um CorroutineScope... aí não sei se vale a pena =(

Reply:
Justo